src/Form/ContactMessageForm.php

```diff
@@ -111,6 +112,11 @@ public function save(array $form, FormStateInterface $form_state): void {
     // The values of the submitted fields.
     $full_view = $this->entityTypeManager->getViewBuilder('contact_message')->view($message, 'full');
     $this->messenger()->addMessage($full_view);
+
+    // Redirect back to same page if redirect value is not set.
+    if (!$contact_form->getRedirectPath()) {
+      $form_state->setRedirectUrl(Url::fromUserInput(\Drupal::service('path.current')->getPath()));
```

Comment:
Dependency injection?

Also, to determine the current path, try to use either `Url::fromRoute('<current>')` or get the current route name and params from the route match (and potential query params from the request) and use Url::fromRoute.

Reply:
Url::fromRoute('<current>') worked, I wont be injecting the service then

Reply:
@drishu And what if the route has/needs params? And query params? Those you get from where? Route match. And that you need to inject. Query params? From request. And that you need to inject :)

Reply:
yes, `<current>` used, my comment was displaying wrong, sry